src/vs/workbench/parts/files/browser/views/explorerView.ts

```diff
@@ -261,6 +264,15 @@ export class ExplorerView extends CollapsibleView {
 			needsRefresh = this.filter.updateConfiguration();
 		}
 
+		// Check if sortOrder value changed
+		const configSortOrder = configuration && configuration.explorer && configuration.explorer.sortOrder;
+		if (this.sortOrder !== configSortOrder) {
+			this.sortOrder = configSortOrder;
+			const sorter = this.getSorter(this.sortOrder);
+			(<Tree>this.explorerViewer).setSorter(sorter);
```

Comment:
Instead of setting a different sorter after the configuration changed, I suggest to actually let the sorter implement the listening and update itself based on the config. You still need this code to decide to do the refresh, which is fine.